junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java

```diff
@@ -86,4 +90,15 @@ void canOverrideAttributeWhenContextIsExtended() {
 		assertSame(testInstanceProvider, newContext.getTestInstanceProvider());
 	}
 
+	@Test
+	void closeAttemptExceptionWillBeThrownDownTheCallStack() throws Exception {
+		ExtensionContext failingExtensionContext = mock(ExtensionContext.class,
+			withSettings().extraInterfaces(AutoCloseable.class));
+		doThrow(Exception.class).when(((AutoCloseable) failingExtensionContext)).close();
+		JupiterEngineExecutionContext newContext = originalContext.extend().withExtensionContext(
+			failingExtensionContext).build();
+
+		assertThrows(Exception.class, newContext::close);
```

Comment:
I think we should check that the thrown exception is the same as the one stubbed above.

Reply:
Done.